[BUG] 500 Internal Server Error when sorting Tags by Document Count

### Description

Clicking the "Document Count" column header on the Tags management page returns a 500 Internal Server Error. The API endpoint /api/tags/?page=1&ordering=document_count crashes with a Django FieldError because the document_count annotation is lost when the queryset is rebuilt for tree node hierarchy filtering.

### Steps to reproduce

1. Navigate to Settings → Tags
2. Click the "Document Count" column header to sort
3. Observe the spinner hang and HTTP 500 response

### Webserver logs

```bash
django.core.exceptions.FieldError: Cannot resolve keyword 'document_count' into field.
Choices are: color, documents, id, is_inbox_tag, is_insensitive, mailrule, match,
matching_algorithm, name, owner, owner_id, tn_ancestors_count, tn_ancestors_pks,
tn_children, tn_children_count, tn_children_pks, tn_depth, tn_descendants_count,
tn_descendants_pks, tn_index, tn_level, tn_order, tn_parent, tn_parent_id,
tn_priority, tn_siblings_count, tn_siblings_pks, workflowtrigger,
workflowtriggers_has_all, workflowtriggers_has_not

File "/usr/src/paperless/src/documents/views.py", line 490, in list
    Tag.objects.filter(pk__in=descendant_pks | {t.pk for t in all_tags})

[2026-03-03 17:06:56,811] [INFO] [celery.worker.consumer.connection] Connected to redis://default:**@redis:6379//
2026-03-04 01:06:56.833251+00:00[2026-03-03 17:06:56,832] [INFO] [celery.apps.worker] celery@3d8f2a619202 ready.
2026-03-04 01:06:56.847383+00:00[2026-03-03 17:06:56,846] [INFO] [celery.worker.strategy] Task documents.tasks.check_scheduled_workflows[39c86aea-1380-4b49-b27b-0add61208f68] received
2026-03-04 01:06:56.851185+00:00[2026-03-03 17:06:56,850] [INFO] [celery.worker.strategy] Task documents.tasks.train_classifier[b98a9022-87a2-4149-bae9-d2c8ac55064d] received
2026-03-04 01:06:56.854255+00:00[2026-03-03 17:06:56,853] [INFO] [celery.worker.strategy] Task paperless_mail.tasks.process_mail_accounts[6d50f512-6b45-4040-bf1a-1a72347ade28] received
2026-03-04 01:06:57.054390+00:00[2026-03-03 17:06:57,053] [INFO] [celery.app.trace] Task documents.tasks.check_scheduled_workflows[39c86aea-1380-4b49-b27b-0add61208f68] succeeded in 0.15728640893939883s: None
2026-03-04 01:06:57.670004+00:00[2026-03-03 17:06:57,669] [INFO] [celery.app.trace] Task paperless_mail.tasks.process_mail_accounts[6d50f512-6b45-4040-bf1a-1a72347ade28] succeeded in 0.07424058392643929s: 'No new documents were added.'
2026-03-04 01:07:04.282206+00:00[2026-03-03 17:07:04,281] [INFO] [paperless.classifier] No updates since last training
2026-03-04 01:07:04.340172+00:00[2026-03-03 17:07:04,339] [INFO] [celery.app.trace] Task documents.tasks.train_classifier[b98a9022-87a2-4149-bae9-d2c8ac55064d] succeeded in 7.3130099709378555s: None
2026-03-04 01:10:00.024551+00:00[2026-03-03 17:10:00,023] [INFO] [celery.beat] Scheduler: Sending due task Check all e-mail accounts (paperless_mail.tasks.process_mail_accounts)
2026-03-04 01:10:00.030206+00:00[2026-03-03 17:10:00,028] [INFO] [celery.worker.strategy] Task paperless_mail.tasks.process_mail_accounts[a54c1bac-ec6b-4582-b270-2fd2e72b7352] received
2026-03-04 01:10:00.140789+00:00[2026-03-03 17:10:00,139] [INFO] [celery.app.trace] Task paperless_mail.tasks.process_mail_accounts[a54c1bac-ec6b-4582-b270-2fd2e72b7352] succeeded in 0.10708110895939171s: 'No new documents were added.'
2026-03-04 01:11:42.737526+00:00/usr/local/lib/python3.12/site-packages/django/core/handlers/asgi.py:332: Warning: StreamingHttpResponse must consume synchronous iterators in order to serve them asynchronously. Use an asynchronous iterator instead.
2026-03-04 01:11:42.737642+00:00async for part in content:
2026-03-04 01:11:50.481140+00:00[2026-03-03 17:11:50,474] [ERROR] [django.request] Internal Server Error: /api/tags/
2026-03-04 01:11:50.481268+00:00Traceback (most recent call last):
2026-03-04 01:11:50.481318+00:00File "/usr/local/lib/python3.12/site-packages/asgiref/sync.py", line 489, in thread_handler
2026-03-04 01:11:50.481394+00:00raise exc_info[1]
2026-03-04 01:11:50.481433+00:00File "/usr/local/lib/python3.12/site-packages/django/core/handlers/exception.py", line 42, in inner
2026-03-04 01:11:50.481474+00:00response = await get_response(request)
2026-03-04 01:11:50.481507+00:00^^^^^^^^^^^^^^^^^^^^^^^^^^^
2026-03-04 01:11:50.481553+00:00File "/usr/local/lib/python3.12/site-packages/asgiref/sync.py", line 489, in thread_handler
2026-03-04 01:11:50.481626+00:00raise exc_info[1]
2026-03-04 01:11:50.481667+00:00File "/usr/local/lib/python3.12/site-packages/django/core/handlers/base.py", line 253, in _get_response_async
2026-03-04 01:11:50.481723+00:00response = await wrapped_callback(
2026-03-04 01:11:50.481759+00:00^^^^^^^^^^^^^^^^^^^^^^^
2026-03-04 01:11:50.481795+00:00File "/usr/local/lib/python3.12/site-packages/asgiref/sync.py", line 439, in __call__
2026-03-04 01:11:50.481837+00:00ret = await asyncio.shield(exec_coro)
2026-03-04 01:11:50.481878+00:00^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2026-03-04 01:11:50.481945+00:00File "/usr/local/lib/python3.12/site-packages/asgiref/current_thread_executor.py", line 40, in run
2026-03-04 01:11:50.481993+00:00result = self.fn(*self.args, **self.kwargs)
2026-03-04 01:11:50.482039+00:00^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2026-03-04 01:11:50.482078+00:00File "/usr/local/lib/python3.12/site-packages/asgiref/sync.py", line 493, in thread_handler
2026-03-04 01:11:50.482144+00:00return func(*args, **kwargs)
2026-03-04 01:11:50.482184+00:00^^^^^^^^^^^^^^^^^^^^^
2026-03-04 01:11:50.482220+00:00File "/usr/local/lib/python3.12/site-packages/django/views/decorators/csrf.py", line 65, in _view_wrapper
2026-03-04 01:11:50.482273+00:00return view_func(request, *args, **kwargs)
2026-03-04 01:11:50.482309+00:00^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2026-03-04 01:11:50.482345+00:00File "/usr/local/lib/python3.12/site-packages/rest_framework/viewsets.py", line 125, in view
2026-03-04 01:11:50.482384+00:00return self.dispatch(request, *args, **kwargs)
2026-03-04 01:11:50.482449+00:00^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2026-03-04 01:11:50.482490+00:00File "/usr/local/lib/python3.12/site-packages/rest_framework/views.py", line 515, in dispatch
2026-03-04 01:11:50.482527+00:00response = self.handle_exception(exc)
2026-03-04 01:11:50.482567+00:00^^^^^^^^^^^^^^^^^^^^^^^^^^
2026-03-04 01:11:50.482631+00:00File "/usr/local/lib/python3.12/site-packages/rest_framework/views.py", line 475, in handle_exception
2026-03-04 01:11:50.482670+00:00self.raise_uncaught_exception(exc)
2026-03-04 01:11:50.482711+00:00File "/usr/local/lib/python3.12/site-packages/rest_framework/views.py", line 486, in raise_uncaught_exception
2026-03-04 01:11:50.482745+00:00raise exc
2026-03-04 01:11:50.482798+00:00File "/usr/local/lib/python3.12/site-packages/rest_framework/views.py", line 512, in dispatch
2026-03-04 01:11:50.482836+00:00response = handler(request, *args, **kwargs)
2026-03-04 01:11:50.482883+00:00^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2026-03-04 01:11:50.482933+00:00File "/usr/src/paperless/src/documents/views.py", line 490, in list
2026-03-04 01:11:50.483003+00:00Tag.objects.filter(pk__in=descendant_pks | {t.pk for t in all_tags})
2026-03-04 01:11:50.483056+00:00File "/usr/local/lib/python3.12/site-packages/django/db/models/query.py", line 1722, in order_by
2026-03-04 01:11:50.483105+00:00obj.query.add_ordering(*field_names)
2026-03-04 01:11:50.483177+00:00File "/usr/local/lib/python3.12/site-packages/django/db/models/sql/query.py", line 2305, in add_ordering
2026-03-04 01:11:50.483225+00:00self.names_to_path(item.split(LOOKUP_SEP), self.model._meta)
2026-03-04 01:11:50.483262+00:00File "/usr/local/lib/python3.12/site-packages/django/db/models/sql/query.py", line 1806, in names_to_path
2026-03-04 01:11:50.483319+00:00raise FieldError(
2026-03-04 01:11:50.483353+00:00django.core.exceptions.FieldError: Cannot resolve keyword 'document_count' into field. Choices are: color, documents, id, is_inbox_tag, is_insensitive, mailrule, match, matching_algorithm, name, owner, owner_id, tn_ancestors_count, tn_ancestors_pks, tn_children, tn_children_count, tn_children_pks, tn_depth, tn_descendants_count, tn_descendants_pks, tn_index, tn_level, tn_order, tn_parent, tn_parent_id, tn_priority, tn_siblings_count, tn_siblings_pks, workflowtrigger, workflowtriggers_has_all, workflowtriggers_has_not
2026-03-04 01:13:32.389926+00:00[2026-03-03 17:13:32,388] [ERROR] [django.request] Internal Server Error: /api/tags/
2026-03-04 01:13:32.390082+00:00Traceback (most recent call last):
2026-03-04 01:13:32.390126+00:00File "/usr/local/lib/python3.12/site-packages/asgiref/sync.py", line 489, in thread_handler
2026-03-04 01:13:32.390203+00:00raise exc_info[1]
2026-03-04 01:13:32.390239+00:00File "/usr/local/lib/python3.12/site-packages/django/core/handlers/exception.py", line 42, in inner
2026-03-04 01:13:32.390276+00:00response = await get_response(request)
2026-03-04 01:13:32.390312+00:00^^^^^^^^^^^^^^^^^^^^^^^^^^^
2026-03-04 01:13:32.390366+00:00File "/usr/local/lib/python3.12/site-packages/asgiref/sync.py", line 489, in thread_handler
2026-03-04 01:13:32.390401+00:00raise exc_info[1]
2026-03-04 01:13:32.390435+00:00File "/usr/local/lib/python3.12/site-packages/django/core/handlers/base.py", line 253, in _get_response_async
2026-03-04 01:13:32.390483+00:00response = await wrapped_callback(
2026-03-04 01:13:32.390520+00:00^^^^^^^^^^^^^^^^^^^^^^^
2026-03-04 01:13:32.390558+00:00File "/usr/local/lib/python3.12/site-packages/asgiref/sync.py", line 439, in __call__
2026-03-04 01:13:32.390603+00:00ret = await asyncio.shield(exec_coro)
2026-03-04 01:13:32.390653+00:00^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2026-03-04 01:13:32.390727+00:00File "/usr/local/lib/python3.12/site-packages/asgiref/current_thread_executor.py", line 40, in run
2026-03-04 01:13:32.390779+00:00result = self.fn(*self.args, **self.kwargs)
2026-03-04 01:13:32.390825+00:00^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2026-03-04 01:13:32.390874+00:00File "/usr/local/lib/python3.12/site-packages/asgiref/sync.py", line 493, in thread_handler
2026-03-04 01:13:32.390942+00:00return func(*args, **kwargs)
2026-03-04 01:13:32.390986+00:00^^^^^^^^^^^^^^^^^^^^^
2026-03-04 01:13:32.391039+00:00File "/usr/local/lib/python3.12/site-packages/django/views/decorators/csrf.py", line 65, in _view_wrapper
2026-03-04 01:13:32.391108+00:00return view_func(request, *args, **kwargs)
2026-03-04 01:13:32.391158+00:00^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2026-03-04 01:13:32.391207+00:00File "/usr/local/lib/python3.12/site-packages/rest_framework/viewsets.py", line 125, in view
2026-03-04 01:13:32.391256+00:00return self.dispatch(request, *args, **kwargs)
2026-03-04 01:13:32.391328+00:00^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2026-03-04 01:13:32.391381+00:00File "/usr/local/lib/python3.12/site-packages/rest_framework/views.py", line 515, in dispatch
2026-03-04 01:13:32.391433+00:00response = self.handle_exception(exc)
2026-03-04 01:13:32.391483+00:00^^^^^^^^^^^^^^^^^^^^^^^^^^
2026-03-04 01:13:32.391563+00:00File "/usr/local/lib/python3.12/site-packages/rest_framework/views.py", line 475, in handle_exception
2026-03-04 01:13:32.391613+00:00self.raise_uncaught_exception(exc)
2026-03-04 01:13:32.391661+00:00File "/usr/local/lib/python3.12/site-packages/rest_framework/views.py", line 486, in raise_uncaught_exception
2026-03-04 01:13:32.391695+00:00raise exc
2026-03-04 01:13:32.391759+00:00File "/usr/local/lib/python3.12/site-packages/rest_framework/views.py", line 512, in dispatch
2026-03-04 01:13:32.391797+00:00response = handler(request, *args, **kwargs)
2026-03-04 01:13:32.391836+00:00^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2026-03-04 01:13:32.391875+00:00File "/usr/src/paperless/src/documents/views.py", line 490, in list
2026-03-04 01:13:32.391925+00:00Tag.objects.filter(pk__in=descendant_pks | {t.pk for t in all_tags})
2026-03-04 01:13:32.391963+00:00File "/usr/local/lib/python3.12/site-packages/django/db/models/query.py", line 1722, in order_by
2026-03-04 01:13:32.391996+00:00obj.query.add_ordering(*field_names)
2026-03-04 01:13:32.392049+00:00File "/usr/local/lib/python3.12/site-packages/django/db/models/sql/query.py", line 2305, in add_ordering
2026-03-04 01:13:32.392098+00:00self.names_to_path(item.split(LOOKUP_SEP), self.model._meta)
2026-03-04 01:13:32.392143+00:00File "/usr/local/lib/python3.12/site-packages/django/db/models/sql/query.py", line 1806, in names_to_path
2026-03-04 01:13:32.392213+00:00raise FieldError(
2026-03-04 01:13:32.392268+00:00django.core.exceptions.FieldError: Cannot resolve keyword 'document_count' into field. Choices are: color, documents, id, is_inbox_tag, is_insensitive, mailrule, match, matching_algorithm, name, owner, owner_id, tn_ancestors_count, tn_ancestors_pks, tn_children, tn_children_count, tn_children_pks, tn_depth, tn_descendants_count, tn_descendants_pks, tn_index, tn_level, tn_order, tn_parent, tn_parent_id, tn_priority, tn_siblings_count, tn_siblings_pks, workflowtrigger, workflowtriggers_has_all, workflowtriggers_has_not
2026-03-04 01:19:24.061187+00:00[2026-03-03 17:19:24,058] [ERROR] [django.request] Internal Server Error: /api/tags/
2026-03-04 01:19:24.061296+00:00Traceback (most recent call last):
2026-03-04 01:19:24.061328+00:00File "/usr/local/lib/python3.12/site-packages/asgiref/sync.py", line 489, in thread_handler
2026-03-04 01:19:24.061381+00:00raise exc_info[1]
2026-03-04 01:19:24.061410+00:00File "/usr/local/lib/python3.12/site-packages/django/core/handlers/exception.py", line 42, in inner
2026-03-04 01:19:24.061433+00:00response = await get_response(request)
2026-03-04 01:19:24.061457+00:00^^^^^^^^^^^^^^^^^^^^^^^^^^^
2026-03-04 01:19:24.061494+00:00File "/usr/local/lib/python3.12/site-packages/asgiref/sync.py", line 489, in thread_handler
2026-03-04 01:19:24.061526+00:00raise exc_info[1]
2026-03-04 01:19:24.061551+00:00File "/usr/local/lib/python3.12/site-packages/django/core/handlers/base.py", line 253, in _get_response_async
2026-03-04 01:19:24.061603+00:00response = await wrapped_callback(
2026-03-04 01:19:24.061626+00:00^^^^^^^^^^^^^^^^^^^^^^^
2026-03-04 01:19:24.061648+00:00File "/usr/local/lib/python3.12/site-packages/asgiref/sync.py", line 439, in __call__
2026-03-04 01:19:24.061671+00:00ret = await asyncio.shield(exec_coro)
2026-03-04 01:19:24.061689+00:00^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2026-03-04 01:19:24.061712+00:00File "/usr/local/lib/python3.12/site-packages/asgiref/current_thread_executor.py", line 40, in run
2026-03-04 01:19:24.061728+00:00result = self.fn(*self.args, **self.kwargs)
2026-03-04 01:19:24.061747+00:00^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2026-03-04 01:19:24.061763+00:00File "/usr/local/lib/python3.12/site-packages/asgiref/sync.py", line 493, in thread_handler
2026-03-04 01:19:24.061786+00:00return func(*args, **kwargs)
2026-03-04 01:19:24.061802+00:00^^^^^^^^^^^^^^^^^^^^^
2026-03-04 01:19:24.061823+00:00File "/usr/local/lib/python3.12/site-packages/django/views/decorators/csrf.py", line 65, in _view_wrapper
2026-03-04 01:19:24.061850+00:00return view_func(request, *args, **kwargs)
2026-03-04 01:19:24.061867+00:00^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2026-03-04 01:19:24.061882+00:00File "/usr/local/lib/python3.12/site-packages/rest_framework/viewsets.py", line 125, in view
2026-03-04 01:19:24.061901+00:00return self.dispatch(request, *args, **kwargs)
2026-03-04 01:19:24.061928+00:00^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2026-03-04 01:19:24.061945+00:00File "/usr/local/lib/python3.12/site-packages/rest_framework/views.py", line 515, in dispatch
2026-03-04 01:19:24.061969+00:00response = self.handle_exception(exc)
2026-03-04 01:19:24.061985+00:00^^^^^^^^^^^^^^^^^^^^^^^^^^
2026-03-04 01:19:24.062007+00:00File "/usr/local/lib/python3.12/site-packages/rest_framework/views.py", line 475, in handle_exception
2026-03-04 01:19:24.062028+00:00self.raise_uncaught_exception(exc)
2026-03-04 01:19:24.062045+00:00File "/usr/local/lib/python3.12/site-packages/rest_framework/views.py", line 486, in raise_uncaught_exception
2026-03-04 01:19:24.062070+00:00raise exc
2026-03-04 01:19:24.062087+00:00File "/usr/local/lib/python3.12/site-packages/rest_framework/views.py", line 512, in dispatch
2026-03-04 01:19:24.062104+00:00response = handler(request, *args, **kwargs)
2026-03-04 01:19:24.062120+00:00^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2026-03-04 01:19:24.062138+00:00File "/usr/src/paperless/src/documents/views.py", line 490, in list
2026-03-04 01:19:24.062162+00:00Tag.objects.filter(pk__in=descendant_pks | {t.pk for t in all_tags})
2026-03-04 01:19:24.062179+00:00File "/usr/local/lib/python3.12/site-packages/django/db/models/query.py", line 1722, in order_by
2026-03-04 01:19:24.062196+00:00obj.query.add_ordering(*field_names)
2026-03-04 01:19:24.062219+00:00File "/usr/local/lib/python3.12/site-packages/django/db/models/sql/query.py", line 2305, in add_ordering
2026-03-04 01:19:24.062239+00:00self.names_to_path(item.split(LOOKUP_SEP), self.model._meta)
2026-03-04 01:19:24.062257+00:00File "/usr/local/lib/python3.12/site-packages/django/db/models/sql/query.py", line 1806, in names_to_path
2026-03-04 01:19:24.062284+00:00raise FieldError(
2026-03-04 01:19:24.062301+00:00django.core.exceptions.FieldError: Cannot resolve keyword 'document_count' into field. Choices are: color, documents, id, is_inbox_tag, is_insensitive, mailrule, match, matching_algorithm, name, owner, owner_id, tn_ancestors_count, tn_ancestors_pks, tn_children, tn_children_count, tn_children_pks, tn_depth, tn_descendants_count, tn_descendants_pks, tn_index, tn_level, tn_order, tn_parent, tn_parent_id, tn_priority, tn_siblings_count, tn_siblings_pks, workflowtrigger, workflowtriggers_has_all, workflowtriggers_has_not
2026-03-04 01:20:00.023943+00:00[2026-03-03 17:20:00,023] [INFO] [celery.beat] Scheduler: Sending due task Check all e-mail accounts (paperless_mail.tasks.process_mail_accounts)
2026-03-04 01:20:00.030057+00:00[2026-03-03 17:20:00,028] [INFO] [celery.worker.strategy] Task paperless_mail.tasks.process_mail_accounts[b26267d8-37ed-46f8-9480-6f2e389a83b0] received
2026-03-04 01:20:00.174251+00:00[2026-03-03 17:20:00,173] [INFO] [celery.app.trace] Task paperless_mail.tasks.process_mail_accounts[b26267d8-37ed-46f8-9480-6f2e389a83b0] succeeded in 0.14031785598490387s: 'No new documents were added.'
2026-03-04 01:30:00.022552+00:00[2026-03-03 17:30:00,021] [INFO] [celery.beat] Scheduler: Sending due task Check all e-mail accounts (paperless_mail.tasks.process_mail_accounts)
2026-03-04 01:30:00.028559+00:00[2026-03-03 17:30:00,027] [INFO] [celery.worker.strategy] Task paperless_mail.tasks.process_mail_accounts[0fcda036-658f-46d6-9912-8b55bb1a6799] received
2026-03-04 01:30:00.186342+00:00[2026-03-03 17:30:00,185] [INFO] [celery.app.trace] Task paperless_mail.tasks.process_mail_accounts[0fcda036-658f-46d6-9912-8b55bb1a6799] succeeded in 0.1539910420542583s: 'No new documents were added.'
2026-03-04 01:40:00.022119+00:00[2026-03-03 17:40:00,021] [INFO] [celery.beat] Scheduler: Sending due task Check all e-mail accounts (paperless_mail.tasks.process_mail_accounts)
2026-03-04 01:40:00.027907+00:00[2026-03-03 17:40:00,026] [INFO] [celery.worker.strategy] Task paperless_mail.tasks.process_mail_accounts[a484d3b9-219a-461b-9eb7-f65042091089] received
2026-03-04 01:40:00.155629+00:00[2026-03-03 17:40:00,154] [INFO] [celery.app.trace] Task paperless_mail.tasks.process_mail_accounts[a484d3b9-219a-461b-9eb7-f65042091089] succeeded in 0.12391118099913001s: 'No new documents were added.'
2026-03-04 01:50:00.020836+00:00[2026-03-03 17:50:00,020] [INFO] [celery.beat] Scheduler: Sending due task Check all e-mail accounts (paperless_mail.tasks.process_mail_accounts)
2026-03-04 01:50:00.026932+00:00[2026-03-03 17:50:00,025] [INFO] [celery.worker.strategy] Task paperless_mail.tasks.process_mail_accounts[00182999-b5ef-47c0-b8dc-1230f6ccab05] received
2026-03-04 01:50:00.137695+00:00[2026-03-03 17:50:00,136] [INFO] [celery.app.trace] Task paperless_mail.tasks.process_mail_accounts[00182999-b5ef-47c0-b8dc-1230f6ccab05] succeeded in 0.10691183398012072s: 'No new documents were added.'
2026-03-04 01:51:17.489284+00:00[2026-03-03 17:51:17,486] [ERROR] [django.request] Internal Server Error: /api/tags/
2026-03-04 01:51:17.489404+00:00Traceback (most recent call last):
2026-03-04 01:51:17.489430+00:00File "/usr/local/lib/python3.12/site-packages/asgiref/sync.py", line 489, in thread_handler
2026-03-04 01:51:17.489451+00:00raise exc_info[1]
2026-03-04 01:51:17.489477+00:00File "/usr/local/lib/python3.12/site-packages/django/core/handlers/exception.py", line 42, in inner
2026-03-04 01:51:17.489509+00:00response = await get_response(request)
2026-03-04 01:51:17.489532+00:00^^^^^^^^^^^^^^^^^^^^^^^^^^^
2026-03-04 01:51:17.489554+00:00File "/usr/local/lib/python3.12/site-packages/asgiref/sync.py", line 489, in thread_handler
2026-03-04 01:51:17.489610+00:00raise exc_info[1]
2026-03-04 01:51:17.489636+00:00File "/usr/local/lib/python3.12/site-packages/django/core/handlers/base.py", line 253, in _get_response_async
2026-03-04 01:51:17.489661+00:00response = await wrapped_callback(
2026-03-04 01:51:17.489683+00:00^^^^^^^^^^^^^^^^^^^^^^^
2026-03-04 01:51:17.489715+00:00File "/usr/local/lib/python3.12/site-packages/asgiref/sync.py", line 439, in __call__
2026-03-04 01:51:17.489737+00:00ret = await asyncio.shield(exec_coro)
2026-03-04 01:51:17.489761+00:00^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2026-03-04 01:51:17.489786+00:00File "/usr/local/lib/python3.12/site-packages/asgiref/current_thread_executor.py", line 40, in run
2026-03-04 01:51:17.489815+00:00result = self.fn(*self.args, **self.kwargs)
2026-03-04 01:51:17.489836+00:00^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2026-03-04 01:51:17.489861+00:00File "/usr/local/lib/python3.12/site-packages/asgiref/sync.py", line 493, in thread_handler
2026-03-04 01:51:17.489884+00:00return func(*args, **kwargs)
2026-03-04 01:51:17.489914+00:00^^^^^^^^^^^^^^^^^^^^^
2026-03-04 01:51:17.489935+00:00File "/usr/local/lib/python3.12/site-packages/django/views/decorators/csrf.py", line 65, in _view_wrapper
2026-03-04 01:51:17.489960+00:00return view_func(request, *args, **kwargs)
2026-03-04 01:51:17.489985+00:00^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2026-03-04 01:51:17.490013+00:00File "/usr/local/lib/python3.12/site-packages/rest_framework/viewsets.py", line 125, in view
2026-03-04 01:51:17.490038+00:00return self.dispatch(request, *args, **kwargs)
2026-03-04 01:51:17.490062+00:00^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2026-03-04 01:51:17.490086+00:00File "/usr/local/lib/python3.12/site-packages/rest_framework/views.py", line 515, in dispatch
2026-03-04 01:51:17.490119+00:00response = self.handle_exception(exc)
2026-03-04 01:51:17.490142+00:00^^^^^^^^^^^^^^^^^^^^^^^^^^
2026-03-04 01:51:17.490166+00:00File "/usr/local/lib/python3.12/site-packages/rest_framework/views.py", line 475, in handle_exception
2026-03-04 01:51:17.490186+00:00self.raise_uncaught_exception(exc)
2026-03-04 01:51:17.490221+00:00File "/usr/local/lib/python3.12/site-packages/rest_framework/views.py", line 486, in raise_uncaught_exception
2026-03-04 01:51:17.490242+00:00raise exc
2026-03-04 01:51:17.490267+00:00File "/usr/local/lib/python3.12/site-packages/rest_framework/views.py", line 512, in dispatch
2026-03-04 01:51:17.490299+00:00response = handler(request, *args, **kwargs)
2026-03-04 01:51:17.490319+00:00^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2026-03-04 01:51:17.490342+00:00File "/usr/src/paperless/src/documents/views.py", line 490, in list
2026-03-04 01:51:17.490366+00:00Tag.objects.filter(pk__in=descendant_pks | {t.pk for t in all_tags})
2026-03-04 01:51:17.490398+00:00File "/usr/local/lib/python3.12/site-packages/django/db/models/query.py", line 1722, in order_by
2026-03-04 01:51:17.490417+00:00obj.query.add_ordering(*field_names)
2026-03-04 01:51:17.490456+00:00File "/usr/local/lib/python3.12/site-packages/django/db/models/sql/query.py", line 2305, in add_ordering
2026-03-04 01:51:17.490479+00:00self.names_to_path(item.split(LOOKUP_SEP), self.model._meta)
2026-03-04 01:51:17.490512+00:00File "/usr/local/lib/python3.12/site-packages/django/db/models/sql/query.py", line 1806, in names_to_path
2026-03-04 01:51:17.490535+00:00raise FieldError(
2026-03-04 01:51:17.490563+00:00django.core.exceptions.FieldError: Cannot resolve keyword 'document_count' into field. Choices are: color, documents, id, is_inbox_tag, is_insensitive, mailrule, match, matching_algorithm, name, owner, owner_id, tn_ancestors_count, tn_ancestors_pks, tn_children, tn_children_count, tn_children_pks, tn_depth, tn_descendants_count, tn_descendants_pks, tn_index, tn_level, tn_order, tn_parent, tn_parent_id, tn_priority, tn_siblings_count, tn_siblings_pks, workflowtrigger, workflowtriggers_has_all, workflowtriggers_has_not
2026-03-04 02:00:00.020442+00:00[2026-03-03 18:00:00,019] [INFO] [celery.beat] Scheduler: Sending due task Check all e-mail accounts (paperless_mail.tasks.process_mail_accounts)
```

### Browser logs

```bash

```

### Services logs

```bash

```

### Paperless-ngx version

2.20.8

### Host OS

TrueNAS SCALE (Linux 6.12.33-production+truenas x86_64)

### Installation method

Docker - official image

### System status

```json

```

### Browser

_No response_

### Configuration changes

_No response_

### Please confirm the following

- [x] I believe this issue is a bug that affects all users of Paperless-ngx, not something specific to my installation.
- [x] This issue is not about the OCR or archive creation of a specific file(s). Otherwise, please see above regarding OCR tools.
- [x] I have already searched for relevant existing issues and discussions before opening this report.
- [x] I have updated the title field above with a concise description.

--- a/src/documents/tests/test_tag_hierarchy.py
+++ b/src/documents/tests/test_tag_hierarchy.py
@@ -147,6 +147,16 @@ class TestTagHierarchy(APITestCase):
         assert serializer.data  # triggers serialization
         assert "document_count_filter" in context
 
+    def test_tag_list_can_order_by_document_count_with_children(self) -> None:
+        self.document.tags.add(self.child)
+
+        response = self.client.get(
+            "/api/tags/",
+            {"ordering": "document_count"},
+        )
+
+        assert response.status_code == 200
+
     def test_cannot_set_parent_to_self(self):
         tag = Tag.objects.create(name="Selfie")
         resp = self.client.patch(

--- a/src/documents/views.py
+++ b/src/documents/views.py
@@ -487,13 +487,13 @@ class TagViewSet(PermissionsAwareDocumentCountMixin, ModelViewSet):
             user = getattr(getattr(self, "request", None), "user", None)
             children_source = list(
                 annotate_document_count_for_related_queryset(
-                    Tag.objects.filter(pk__in=descendant_pks | {t.pk for t in all_tags})
-                    .select_related("owner")
-                    .order_by(*ordering),
+                    Tag.objects.filter(
+                        pk__in=descendant_pks | {t.pk for t in all_tags},
+                    ).select_related("owner"),
                     through_model=self.document_count_through,
                     related_object_field=self.document_count_source_field,
                     user=user,
-                ),
+                ).order_by(*ordering),
             )
         else:
             children_source = all_tags

